File: Data Structures & Algorithms/surrounded-regions/submission-1.cpp
class Solution {
public:
    void dfs(vector<vector<char>>& board, int row, int col){
        //Check if out of bounds
        if(row < 0 || row >= board.size() || col < 0 || col >= board[0].size() || board[row][col] != 'O'){
            return;
        }

        board[row][col] = '#';

        dfs(board, row + 1, col);
        dfs(board, row - 1, col);
        dfs(board, row, col + 1);
        dfs(board, row, col - 1);

    }
    void solve(vector<vector<char>>& board) {
        for(int row = 0; row < board.size(); row++){
            for(int col = 0; col < board[0].size(); col++){
                if((row == 0 || row == board.size() - 1 || col == 0 || col == board[0].size() - 1) && board[row][col] == 'O'){
                    dfs(board, row, col);
                }
            }
        }

        for(int row = 0; row < board.size(); row++){
            for(int col = 0; col < board[0].size(); col++){
                if(board[row][col] == 'O'){
                    board[row][col] = 'X';
                } else if(board[row][col] == '#'){
                    board[row][col] = 'O';
                }
            }
        }
    }
};

File: Data Structures & Algorithms/search-for-word/submission-1.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        
        def dfs(row, col, wordIndex):
            if wordIndex == len(word):
                return True
            
            if row < 0 or row > len(board) - 1 or col < 0 or col > len(board[0]) - 1 or word[wordIndex] != board[row][col]:
                return False
            
            letter = board[row][col]
            board[row][col] = "#"

            res = dfs(row + 1, col, wordIndex + 1) or dfs(row - 1, col, wordIndex + 1) or dfs(row, col + 1, wordIndex + 1) or dfs(row, col - 1, wordIndex + 1)

            board[row][col] = letter
            return res

        for row in range(len(board)):
            for col in range(len(board[0])):
                if board[row][col] == word[0] and dfs(row, col, 0):
                    return True
        
        return False